pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -197,6 +198,7 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 			PublicAddress: publicAddress,
 			ReadWritePort: port,
 
+			Authenticator:    originAuthenticator, // this is used to fulfill the tokenreviews endpoint which is used by node authentication
```

Comment:
Looks like we ignore the upstream handler chain, otherwise impersonation would fail